core/src/main/java/org/apache/druid/java/util/emitter/service/SegmentMetadataEvent.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.java.util.emitter.service;
+
+import com.fasterxml.jackson.annotation.JsonValue;
+import com.google.common.collect.ImmutableMap;
+import org.apache.druid.java.util.emitter.core.Event;
+import org.joda.time.DateTime;
+
+import java.util.Map;
+
+public class SegmentMetadataEvent implements Event
+{
+  public static final String FEED = "feed";
+  public static final String DATASOURCE = "datasource";
+  public static final String CREATED_TIME = "createdTime";
+  public static final String START_TIME = "startTime";
+  public static final String END_TIME = "endTime";
+  public static final String VERSION = "version";
+  public static final String IS_COMPACTED = "isCompacted";
+
+  private DateTime createdTime;
+  private String datasource;
+  private DateTime startTime;
+  private DateTime endTime;
+  private String version;
+  private Boolean isCompacted;
+
+  public SegmentMetadataEvent(
+      String datasource,
+      DateTime createdTime,
+      DateTime startTime,
+      DateTime endTime,
+      String version,
+      Boolean isCompacted
+  )
+  {
+    this.datasource = datasource;
+    this.createdTime = createdTime;
+    this.startTime = startTime;
+    this.endTime = endTime;
+    this.version = version;
+    this.isCompacted = isCompacted;
+  }
+
+  @Override
+  public String getFeed()
+  {
+    return "segment_metadata";
+  }
+
+  @Override
+  @JsonValue
+  public Map<String, Object> toMap()
```

Comment:
any particular reason we are using this peculiar way of serializing this object to json instead of the usual way using annotations for all the fields, e.g. https://github.com/apache/druid/blob/master/core/src/main/java/org/apache/druid/indexer/TaskStatus.java#L98-L128

Reply:
This `toMap()` method is needed to implement the event interface. 
This function basically converts the event to a Map
Some references:
[AlertEvent](https://github.com/apache/druid/blob/master/core/src/main/java/org/apache/druid/java/util/emitter/service/AlertEvent.java#L152)
[ServiceMetricEvent](https://github.com/apache/druid/blob/master/core/src/main/java/org/apache/druid/java/util/emitter/service/ServiceMetricEvent.java#L110)